src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -993,6 +992,313 @@ public static boolean removeConnectedRouting(Net net, Node node){
+	            updateFanout.add(startNode);
+	            if (new Node(pip.getTile(), pip.getStartWireIndex()).equals(sink) && !atReversedBidirectionalPip) {
+	                // reached the source and there is another branch starting with a reversed
+	                // bidirectional PIP ... don't traverse it
+	                break;
+	            }
+	            sink = new Node(pip.getTile(), atReversedBidirectionalPip ? pip.getEndWireIndex() :
+	                pip.getStartWireIndex());
+	            curr = reverseConns.get(sink);
+	            if(curr != null && curr.size() > 1 && atReversedBidirectionalPip) {
+	                for(PIP reversePIP : toRemove) {
+	                    curr.remove(reversePIP);
+	                }
+	            }
+	            atReversedBidirectionalPip = false;
+	            fanoutCount = fanout.get(sink);
+	            fanoutCount = fanoutCount == null ? 0 : fanoutCount;
+	            SitePin sitePin = sink.getSitePin();
+	            if (curr == null && !(sitePin != null || sink.getWireName().contains(Net.VCC_WIRE_NAME) ||
+	                    sink.getWireName().contains(Net.GND_WIRE_NAME))) {
+	                // curr should only be null when we're at the source site, so we've hit a reversed bidirectional PIP
+	                // on our linear path
+	                curr = reverseConnsStart.get(sink);
+	                curr.remove(pip);
+	                fanoutCount--;
+	                atReversedBidirectionalPip = true;
+	            }
+	            if(sitePin != null && sitePin.isInput()){
+	                SiteInst si = p.getSiteInst().getDesign().getSiteInstFromSite(sitePin.getSite());
+	                if(si != null){
+	                    if(net.equals(si.getNetFromSiteWire(sitePin.getPinName()))){
+	                        fanoutCount = 2;
+	                    }
+	                }
+	            }
+	        }
+	        if(curr == null && fanout.size() == 1 && !net.isStaticNet()){
+	            // We got all the way back to the source site. It is likely that 
+	            // the net is using dual exit points from the site as is common in
+	            // SLICEs -- we should unroute the sitenet
+	            SitePin sPin = sink.getSitePin();
+	            SiteInst si = net.getSource().getSiteInst();
+	            BELPin belPin = sPin.getBELPin();
+	            si.unrouteIntraSiteNet(belPin, belPin);
+	        }
+	        p.setRouted(false);	        
+	        for(Node startNode : updateFanout) {
+	            Integer newFanout = fanout.get(startNode);
+	            if(newFanout != null) {
+	                newFanout--;
+	                fanout.put(startNode, newFanout);
+	            }
+	        }
+	    }
+	    ArrayList<PIP> updatedPIPs = new ArrayList<>();
+	    for(PIP pip : net.getPIPs()){
+	        if(!toRemove.contains(pip)) updatedPIPs.add(pip);
+	    }
+	    net.setPIPs(updatedPIPs);
+	}
+
+
+	/**
+	 * This method will completely remove a placed cell (both logical and physical) from a design.
```

Comment:
```suggestion
	 * This method will completely remove a placed cell (both logical and physical) from a design.
	 * In the case where the removed cell is the last user of a shared control signal (CLK, CE, SR) then that pin will also be removed and unrouted immediately if deferRemovals is null, otherwise it is added to this map.
```